### PR TITLE
[Feature/#275] animated topAppBar

### LINKED ISSUE #275: Centering the app bar on the About list screen when scrolling
## Overview 
This Issue is related to #223, PR  #270.

According to [figma](https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI?node-id=55725-8670&t=htgSgF3wWucKlsEr-0), The text on the TopAppBar on the About List screen is located on the left edge.
image

**However**, the text on the TopAppBar must be centered on the screen when **scrolled** ([figma](https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI?node-id=54982-75342&t=htgSgF3wWucKlsEr-0))

image

What we will probably have to do is
1. detect scrolling
2. center the text
    - Ideally, use animation to make it move smoothly

### PR DESCRIPTION
**Pull request by chattymin**
## Issue
- close #275 

## Overview (Required)
- Added `AnimatedTextTopAppBar` and apply `about feature`
- Apply [figma](https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI?node-id=54795-26746&t=wUtMwRZ6sH0KjDcQ-0)

## Links
- 

## Movie (Optional)
Before | After
:--: | :--:
<video src="https://github.com/user-attachments/assets/a599b6b7-9f6b-4863-9682-2da3c27e7da4" width="300" > | <video src="https://github.com/user-attachments/assets/82299b7b-3768-4af5-a38d-a203ec2377bc" width="300" >

### CODE REVIEW

File: core/designsystem/src/commonMain/kotlin/io/github/droidkaigi/confsched/designsystem/component/AnimatedTextTopAppBar.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.TopAppBarScrollBehavior
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.BiasAlignment
+import androidx.compose.ui.Modifier
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AnimatedTextTopAppBar(
```
Comment (chattymin) [inline core/designsystem/src/commonMain/kotlin/io/github/droidkaigi/confsched/designsystem/component/AnimatedTextTopAppBar.kt:23]: This component is similar to issue #228. However, this feature requires the use of TopAppBar, while issue #288 requires LargeTopAppBar, making it difficult to use a common component for both. Therefore, I implemented it separately and submitted a PR.

**Comment by takahirom** [comment]
Thank you for your contribution. The animation is awesome. However, our designer prefers the #292 animation. Could you use AnimatedLargeTopAppBar instead?
https://github.com/DroidKaigi/conference-app-2024/pull/292#issuecomment-2282900929

**Comment by chattymin** [comment]
> Thank you for your contribution. The animation is awesome. However, our designer prefers the #292 animation. Could you use AnimatedLargeTopAppBar instead? [#292 (comment)](https://github.com/DroidKaigi/conference-app-2024/pull/292#issuecomment-2282900929)

Sure! However, #292 uses a LargeTopAppBar with an icon at the top, while #293 uses a TopAppBar without an icon. If we use the AnimatedLargeTopAppBar as is, it will include a back icon at the top. Even if we set the icon to be optional and hide it, there will still be unnecessary padding.

So, instead of using AnimatedLargeTopAppBar, I used a custom component that I created and added the same fade-in and fade-out animations as in #292. What do you think about this approach? @takahirom

**Comment by takahirom** [comment]
Sounds good! That would be appreciated!

**Comment by takahirom** [comment]
Thank you for your contribution!

**Pull request merged**